satellite/console/usercredits.go

```diff
@@ -53,3 +54,39 @@ type UserCreditUsage struct {
 	AvailableCredits currency.USD
 	UsedCredits      currency.USD
 }
+
+// CreateCredit holds information that's needed when create a new record of user credit
+type CreateCredit struct {
+	OfferInfo     rewards.RedeemOffer
+	UserID        uuid.UUID
+	OfferID       int
+	Type          CreditType
+	ReferredBy    *uuid.UUID
+	CreditsEarned currency.USD
+	ExpiresAt     time.Time
+}
+
+// NewCredit returns a new credit data
+func NewCredit(currentReward *rewards.Offer, creditType CreditType, userID uuid.UUID, referrerID *uuid.UUID) (*CreateCredit, error) {
+	var creditEarned currency.USD
+	switch creditType {
+	case Invitee:
+		creditEarned = currency.Cents(0)
```

Comment:
we want to do 0 ?

Reply:
I was debating whether we should have a parameter to pass in the number to make it more flexible. It's 0 here because we can only give invitee credits when they activate their account

Reply:
Added a comment